components/21-atoms/table/table.functions.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+var ResponsiveTable = function (element, options) { // eslint-disable-line no-unused-vars
+  /**
+   * The table wrapper.
+   */
+  var table = null;
+
+  /**
+   * The table caption.
+   */
+  var caption = null;
+
+  /**
+   * The options for the table.
+   */
+  var opts = {};
+
+  /**
+   * Determine if the table should be focusable.
+   */
+  var determineFocusable = function () {
+    var scrollableWidth = table.parentNode.querySelector('table').scrollWidth; // eslint-disable-line no-console
+    var containerWidth = table.parentNode.clientWidth; // eslint-disable-line no-console
+
+    // Check if element is scrollable.
+    if (scrollableWidth <= containerWidth) {
+      // If not remove the tab focus.
+      table.removeAttribute('tabindex');
+    }
+    else {
+      caption.innerText = caption.innerText + ' ' + opts.scrollableText;
+    }
+  };
+
+  /**
+   * Setup the responsive table.
+   */
+  var setupResponsiveTable = function () {
+    addTableConotainerAttributes();
+    addCaptionAttribute();
```

Comment:
I don't really see an added value in the separate functions `addTableConotainerAttributes` and `addCaptionAttribute`. It would improve readability if you just specified the contents inside the current function. 

(Unless you have a solid reason to create separate functions, of course)

Reply:
Good point it might have been a bit unnecessary :P